src/comodo/robotModel/robotModel.py

```diff
@@ -121,13 +125,16 @@ def compute_desired_position_walking(self):
         H_torso = self.w_H_torso(H_b, self.s)
         quat_torso = self.rotation_matrix_to_quaternion(H_torso[:3, :3])
         reference_rotation = np.asarray([1.0, 0.0, 0.0, 0.0])
-        self.solver.subject_to(self.s[17] == desired_knee)
-        self.solver.subject_to(self.s[11] == desired_knee)
-        self.solver.subject_to(self.s[3] == elbow)
-        self.solver.subject_to(self.s[7] == elbow)
-        self.solver.subject_to(self.s[1] == shoulder_roll)
-        self.solver.subject_to(self.s[5] == shoulder_roll)
-        self.solver.subject_to(self.s[9] == self.s[15])
+        for index, joint_name in enumerate(self.joint_name_list):
+            if "knee" in joint_name:
+                self.solver.subject_to(self.s[index] == desired_knee)
+            if "shoulder_roll" in joint_name: 
+                self.solver.subject_to(self.s[index] == shoulder_roll)
+            if "elbow" in joint_name: 
+                self.solver.subject_to(self.s[index] == elbow)    
```

Comment:
General comment, the class name is a generic "robotModel", but these lines are very specific to the ergoCub/iCub robots

Reply:
Yes I agree, I think once we have merged the major PRs, we should work on a generalization of the robot model. I would do it in a second step